analysis/deep2_r23_o32/zoom_and_gauss_general.py

```diff
@@ -198,235 +214,203 @@ def zoom_gauss_plot(dataset, fitspath, tab, stack2D, dispersion, s2, wave,
-                    txt0 += 'RMS: %.3f RMS/pix: %.3f, N: %i \n' % (RMS_tot, RMS_pix, N_gal_array[rr])
-                    txt0 += r'Median: %.3f $\sigma$: %.3f  Norm: %.3f o1[2]: %.3f'% (o1[3], o1[1], max0,o1[2]) + '\n'
-                    txt0 += 'Flux_G: %.3f Flux_S: %.3f' %(flux_g, flux_s) + '\n'
-                    txt0 += 'S/N: %.3f' %(SN_array[rr])
+                    txt0 += 'RMS: %.3f RMS/pix: %.3f, N: %i \n' % (rms_tot, rms_pix, N_gal_array[rr])
+                    txt0 += r'Median: %.3f $\sigma$: %.3f  Norm: %.3f o1[2]: %.3f' % (o1[3], o1[1], max0, o1[2]) + '\n'
+                    txt0 += 'Flux_G: %.3f Flux_S: %.3f' % (flux_g, flux_s) + '\n'
+                    txt0 += 'S/N: %.3f' % (SN_array[rr])
 
             
-            #t_ax.annotate(txt0, [0.95,0.95], xycoords='axes fraction', va='top', ha='right',fontsize= '5')
-            for x in  lambda0: t_ax.axvline(x=x, linewidth= 0.15, color= 'k', linestyle = '--')
+            # t_ax.annotate(txt0, [0.95,0.95], xycoords='axes fraction', va='top', ha='right',fontsize= '5')
+
+            for x in lambda0:
+                t_ax.axvline(x=x, linewidth=0.15, color='k', linestyle='--')
 
-            txt1 = 'Intensity' #Units: ' + r'$ergs *s^{-1} *cm^{-2} *\AA$' + '\n'
-            #txt1 += 'Scale factor = 1e-17'
+            txt1 = 'Intensity'  # Units: ' + r'$ergs *s^{-1} *cm^{-2} *\AA$' + '\n'
             
             if col == 0:
                 t_ax.set_ylabel(txt1)
-            else: t_ax.set_yticklabels([])  # sets y-tick labels
+            else:
+                t_ax.set_yticklabels([])  # sets y-tick labels
 
-            if row != nrows-1 and rr != stack2D.shape[0]-1:
+            if row != nrows-1 and rr != stack2d.shape[0]-1:
                 t_ax.set_xticklabels([])
         else:
-            t_ax.plot(wave, y_norm,'k', linewidth=0.3, label= 'Emission')
-            t_ax.set_xlim(x1+50,x2-50)
-            
-            
-        if (rr % (nrows*ncols) == nrows*ncols-1) or rr == stack2D.shape[0]-1: 
+            t_ax.plot(wave, y_norm, 'k', linewidth=0.3, label='Emission')
+            t_ax.set_xlim(x1 + 50, x2 - 50)
+
+        if (rr % (nrows * ncols) == nrows * ncols - 1) or rr == stack2d.shape[0] - 1:
             subplots_adjust(left=0.1, right=0.98, bottom=0.06, top=0.97, hspace=0.05)
 
-            fig.set_size_inches(8,8)
+            fig.set_size_inches(8, 8)
             fig.savefig(pdf_pages, format='pdf')
             fig.clear()
     # endfor
 
-
     # Writing Ascii Tables and Fits Tables
     # Repetive Columns have been added: The last four ot six columns will be used for individual graphing
-    out_ascii = fitspath+'/'+dataset+'_flux_gaussian_'+str(np.int(working_wave))+'.tbl'
-    # out_fits = fitspath+'/'+dataset+'_Flux_gaussian_'+line_name+'.fits'
-
+    out_ascii = fitspath+'/' + dataset + '_flux_gaussian_' + str(np.int(working_wave)) + '.tbl'
```

Comment:
```suggestion
    out_ascii = join(fitspath, f'{dataset}_flux_gaussian_{str(np.int(working_wave))}.tbl'
```

Reply:
This is using f-strings.  Hopefully you have v3.6+ for Python.  It's easier to read and require less whitespace.

Reply:
thanks